NeoML/src/TraditionalML/FloatVector.cpp

```diff
@@ -49,23 +57,27 @@ CFloatVector::CFloatVector( int size, const CSparseFloatVectorDesc& desc )
 {
 	NeoAssert( size >= 0 );
 	auto bodyPtr = FINE_DEBUG_NEW CFloatVectorBody( size );
-	int ptrPos = 0;
-	int ptrSize = desc.Size;
 
-	float value = 0;
-	for( int i = 0; i < size; i++ ) {
-		if( ptrPos >= ptrSize || i < desc.Indexes[ptrPos] ) {
-			value = 0;
-		} else {
-			value = desc.Values[ptrPos];
-			ptrPos++;
+	if( desc.Indexes == nullptr ) {
+		NeoAssert( size == desc.Size );
```

Comment:
Why?

Reply:
because in sparse version "No elements should stay unprocessed!"

Reply:
NeoAssert( size >= desc.Size ); ?